perllib/FixMyStreet/Script/Questionnaires.pm

```diff
@@ -43,6 +43,9 @@ sub send_questionnaires_period {
 
     while (my $row = $unsent->next) {
 
+        # User may have opted out of questionnaires
+        next unless $row->user->receive_questionnaires;
```

Comment:
We need to mark the report as not sending questionnaires. Otherwise if in 6 months they turn getting questionnaires back on, they will presumably receive every questionnaire they've ever missed :)
If you move this down to the check around line 63 to add to the set of checks there (cobrand_send, row->is_closed, etc) and add this as an extra check there.